refactor(accounts-management)_: break dependency on the eth-node types

saledjenic opened the pull request:
This commit introduces a new `crypto` package to handle cryptographic operations and types. The changes aim to break
dependency on `eth-node`, improve maintainability and usability of the accounts management system.

- Add new crypto package with interface-based design following keystore pattern
- Create crypto/types package with Address, HexBytes, and Hash types
- Move core cryptographic types from eth-node to accounts-management/crypto/types
- Update core interfaces (keystore, persistence) to use local types
- Implement GethCryptoProvider using go-ethereum crypto functions
- Update generator and common packages to use new crypto package
- Enable self-contained accounts-management package structure

Note: the last commit contains code updates due to changes made in previous commits

saledjenic committed:
refactor(accounts-management)_: simplify geth keystore adapter initialization

The `NewGethKeystoreAdapter` function has been updated to remove the scrypt parameters, and keep them internal to the adapter part.

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (18)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 3db1da58 | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6825/1/) | 2025-08-11 21:44:28 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250811-214159-3db1da-pr6825.aar) |
| :heavy_check_mark: | 3db1da58 | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6825/1/) | 2025-08-11 21:44:55 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250811-214202-3db1da-pr6825.zip) |
| :heavy_check_mark: | 3db1da58 | [#1](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6825/1/) | 2025-08-11 21:45:13 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250811-214206-3db1da-pr6825.zip) |
| :heavy_check_mark: | 3db1da58 | [#1](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6825/1/) | 2025-08-11 21:46:15 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250811-214209-3db1da-pr6825.zip) |
| :heavy_check_mark: | 3db1da58 | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6825/1/) | 2025-08-11 21:47:36 | ~5 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250811-214200-3db1da-pr6825.zip) |
| :heavy_check_mark: | 3db1da58 | [#1](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6825/1/) | 2025-08-11 21:48:16 | ~6 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250811-214213-3db1da-pr6825.zip) |
| :heavy_check_mark: | 3db1da58 | [#1](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6825/1/) | 2025-08-11 21:50:07 | ~7 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6825/1//consoleText) |
| :heavy_check_mark: | 3db1da58 | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6825/1/) | 2025-08-11 21:52:19 | ~10 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250811-214203-3db1da-pr6825.zip) |
| :heavy_check_mark: | 3db1da58 | [#1](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6825/1/) | 2025-08-11 22:05:23 | ~23 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6825/1//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 11642d5d | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6825/2/) | 2025-08-12 05:58:40 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250812-055532-11642d-pr6825.zip) |
| :heavy_check_mark: | 11642d5d | [#2](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6825/2/) | 2025-08-12 05:58:49 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250812-055536-11642d-pr6825.zip) |
| :heavy_check_mark: | 11642d5d | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6825/2/) | 2025-08-12 05:59:02 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250812-055530-11642d-pr6825.aar) |
| :heavy_check_mark: | 11642d5d | [#2](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6825/2/) | 2025-08-12 05:59:36 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250812-055540-11642d-pr6825.zip) |
| :heavy_check_mark: | 11642d5d | [#2](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6825/2/) | 2025-08-12 06:01:28 | ~5 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250812-055545-11642d-pr6825.zip) |
| :heavy_check_mark: | 11642d5d | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6825/2/) | 2025-08-12 06:02:28 | ~6 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250812-055531-11642d-pr6825.zip) |
| :heavy_check_mark: | 11642d5d | [#2](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6825/2/) | 2025-08-12 06:03:14 | ~7 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6825/2//consoleText) |
| :heavy_check_mark: | 11642d5d | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6825/2/) | 2025-08-12 06:03:32 | ~7 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250812-055533-11642d-pr6825.zip) |
| :heavy_check_mark: | 11642d5d | [#2](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6825/2/) | 2025-08-12 06:19:15 | ~23 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6825/2//consoleText) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 8c098b77 | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6825/3/) | 2025-08-13 09:57:15 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250813-095424-8c098b-pr6825.aar) |
| :heavy_check_mark: | 8c098b77 | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6825/3/) | 2025-08-13 09:57:23 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250813-095427-8c098b-pr6825.zip) |
| :heavy_check_mark: | 8c098b77 | [#3](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6825/3/) | 2025-08-13 09:57:45 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250813-095431-8c098b-pr6825.zip) |
| :heavy_check_mark: | 8c098b77 | [#3](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6825/3/) | 2025-08-13 09:58:37 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250813-095436-8c098b-pr6825.zip) |
| :heavy_check_mark: | 8c098b77 | [#3](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6825/3/) | 2025-08-13 10:00:19 | ~5 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250813-095438-8c098b-pr6825.zip) |
| :heavy_check_mark: | 8c098b77 | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6825/3/) | 2025-08-13 10:01:17 | ~6 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250813-095426-8c098b-pr6825.zip) |
| :heavy_check_mark: | 8c098b77 | [#3](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6825/3/) | 2025-08-13 10:01:43 | ~7 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6825/3//consoleText) |
| :heavy_check_mark: | 8c098b77 | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6825/3/) | 2025-08-13 10:04:13 | ~9 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250813-095428-8c098b-pr6825.zip) |
| :heavy_check_mark: | 8c098b77 | [#3](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6825/3/) | 2025-08-13 10:17:48 | ~23 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6825/3//consoleText) |
| :heavy_check_mark: | 8c098b77 | [#4](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6825/4/) | 2025-08-13 12:57:29 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250813-125422-8c098b-pr6825.zip) |
| :heavy_check_mark: | 8c098b77 | [#4](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6825/4/) | 2025-08-13 12:57:51 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250813-125415-8c098b-pr6825.aar) |
| :heavy_check_mark: | 8c098b77 | [#4](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6825/4/) | 2025-08-13 12:58:23 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250813-125424-8c098b-pr6825.zip) |
| :heavy_check_mark: | 8c098b77 | [#4](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6825/4/) | 2025-08-13 12:59:44 | ~5 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250813-125428-8c098b-pr6825.zip) |
| :heavy_check_mark: | 8c098b77 | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6825/4/) | 2025-08-13 13:00:18 | ~6 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250813-125417-8c098b-pr6825.zip) |
| :heavy_check_mark: | 8c098b77 | [#4](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6825/4/) | 2025-08-13 13:01:46 | ~7 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250813-125416-8c098b-pr6825.zip) |
| :heavy_multiplication_x: | 8c098b77 | [#4](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6825/4/) | 2025-08-13 13:05:53 | ~11 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6825/4//consoleText) |
| :heavy_check_mark: | 8c098b77 | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6825/4/) | 2025-08-13 13:07:56 | ~13 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250813-125419-8c098b-pr6825.zip) |
| :heavy_check_mark: | 8c098b77 | [#4](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6825/4/) | 2025-08-13 13:21:59 | ~27 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6825/4//consoleText) |
| | | | | | | |
| :heavy_check_mark: | f99dbb38 | [#5](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6825/5/) | 2025-08-13 13:43:50 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250813-134113-f99dbb-pr6825.aar) |
| :heavy_check_mark: | f99dbb38 | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6825/5/) | 2025-08-13 13:44:41 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250813-134116-f99dbb-pr6825.zip) |
| :heavy_check_mark: | f99dbb38 | [#5](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6825/5/) | 2025-08-13 13:44:52 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250813-134123-f99dbb-pr6825.zip) |
| :heavy_check_mark: | f99dbb38 | [#5](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6825/5/) | 2025-08-13 13:45:25 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250813-134127-f99dbb-pr6825.zip) |
| :heavy_check_mark: | f99dbb38 | [#5](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6825/5/) | 2025-08-13 13:47:17 | ~5 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250813-134132-f99dbb-pr6825.zip) |
| :heavy_check_mark: | f99dbb38 | [#5](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6825/5/) | 2025-08-13 13:47:49 | ~6 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250813-134114-f99dbb-pr6825.zip) |
| :heavy_check_mark: | f99dbb38 | [#5](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6825/5/) | 2025-08-13 13:49:05 | ~7 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6825/5//consoleText) |
| :heavy_check_mark: | f99dbb38 | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/nwaku/job/PR-6825/5/) | 2025-08-13 13:49:59 | ~8 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250813-134118-f99dbb-pr6825.zip) |
| :heavy_check_mark: | f99dbb38 | [#5](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6825/5/) | 2025-08-13 14:06:13 | ~24 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6825/5//consoleText) |

osmaczko reviewed (commented):
As discussed offline, crypto is not solely related to account management. It would be more appropriate to place it in `status-go/crypto` or `status-go/internal/crypto` rather than under `status-go/accounts-management/crypto`.

saledjenic committed:
feat(crypto)_: introduce comprehensive cryptographic package

This commit adds a new crypto package that provides a wide range of cryptographic operations tailored for Ethereum compatibility. Key features include:

- Implementation of cryptographic primitives such as key generation, signing, hashing, and address derivation.
- Introduction of a `CryptoProvider` interface for pluggable cryptographic implementations, with a default provider based on Geth.
- Support for Ethereum-specific operations through the `EthereumCrypto` implementation.
- Type safety with strongly typed cryptographic types (Address, Hash, etc.).
- Detailed README documentation for usage and examples.

This package aims to enhance the security and functionality of cryptographic operations within the Status Go application.

saledjenic committed:
refactor(accounts-management)_: update address handling and keystore integration

This commit refines the accounts management package by transitioning from the previous eth-node types to a new cryptographic types structure. Key changes include:

- Replacing `ethtypes.Address` with `cryptotypes.Address` across various components, enhancing type safety and consistency.
- Updating the `KeystoreAccount` type to utilize the new address structure.
- Modifying the keystore interface and related functions to align with the new address type.

These updates aim to improve the maintainability and clarity of the accounts management logic, ensuring better integration with the new cryptographic package.

saledjenic committed:
refactor(accounts-management)_: eth-node related packages removed and the code updated accordingly

- Affected code updated based on the previous commit and removal of eth-node packages.
- Updates to the `api`, `protocol`, and `services` packages to streamline functionality and reduce complexity.
- Refactoring of cryptographic operations and types to enhance usability.
- Cleanup of unused code and dependencies to optimize performance.

saledjenic commented:
> As discussed offline, crypto is not solely related to account management. It would be more appropriate to place it in `status-go/crypto` or `status-go/internal/crypto` rather than under `status-go/accounts-management/crypto`.

We discussed this previously in a private chat. The reason I initially placed the `crypto` package under `accounts-management` was because I had planned to move `accounts-management` into a separate repository. Since both accs-generator and accs-keystore depend on `crypto` functions, my goal was to make that functionality accessible through `accounts-management` so other parts of `status-go` could consume it indirectly.

@osmaczko I agree, however, that `crypto` is not logically tied to account management. Now that we’ve decided not to move `accounts-management` into a separate repository, I’ve pulled out the `crypto` package from `accounts-management`. Otherwise, we would have to move it to its own repository and import it into `accounts-management` separately.
